addons/stock/models/stock_picking.py

```diff
@@ -693,7 +693,7 @@ def _create_link_for_product(operation_id, product_id, qty):
 
         need_rereserve = False
         # sort the operations in order to give higher priority to those with a package, then a lot/serial number
-        operations = self.pack_operation_ids
+        operations = self.pack_operation_ids.sorted('product_qty')
         operations = sorted(operations, key=lambda x: ((x.package_id and not x.product_id) and -4 or 0) + (x.package_id and -2 or 0) + (x.pack_lot_ids and -1 or 0))
```

Comment:
Indeed, to avoid a cumbersome sorting function I am relying on the stability of the CPython's implementation. Since this part is quite touchy, and both v10 and Python 2 will reach their EOL by the end of the year, that seemed to be the best compromise.
But I agree, a clear comment should be added.